recipe/meta.yaml

```
@@ -1,54 +1,168 @@
-    - pytest ./tests/ -v --ignore-glob=*Jax* --ignore-glob=*pytorch*  # [win]
-  requires:
-    - pip
-    - pytest
-    - icub-models
-    - gym-ignition-models
-    - pytorch  # [not win]
-    - jax  # [not win]
-    - idyntree
-    - git
-    - gitpython
+    - python
+
+outputs:
+  - name: {{ name|lower }}
+    build:
+      noarch: python
+      script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+    requirements:
+      build:
+        - python  {{ python_min }}
+      host:
+        - python  {{ python_min }}
+        - pip
+        - wheel
+        - setuptools
+      run:
+        - python  >={{ python_min }}
+        - numpy
+        - scipy
+        - prettytable
+        - urdfdom-py
+    test:
+      source_files:
+        - tests
+      commands:
+        - pytest ./tests/ -v -k "not gravity" --ignore-glob=*Jax* --ignore-glob=*pytorch* --ignore-glob=*CasADi*  --ignore-glob=*idyntree*
+      requires:
+        - python  >={{ python_min }}
+        - pip
+        - pytest
+        - icub-models
+        - gym-ignition-models
+        - idyntree
+        - git
+        - gitpython
+
+  - name: {{ name|lower }}-casadi
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python  {{ python_min }}
+      run:
+        - {{ pin_subpackage('adam-robotics', max_pin='x.x.x') }}
+        - casadi
+        - python  >={{ python_min }}
+    test:
+      source_files:
+        - tests
+      commands:
+        - pytest ./tests/ -v -k "not gravity" --ignore-glob=*Jax* --ignore-glob=*pytorch*
+      requires:
+        - python  >={{ python_min }}
+        - pip
+        - pytest
+        - icub-models
+        - gym-ignition-models
+        - idyntree
+        - git
+        - gitpython
+        - casadi
+
+  - name: {{ name|lower }}-jax
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python  {{ python_min }}
+      run:
+        - {{ pin_subpackage('adam-robotics', max_pin='x.x.x') }}
+        - jax
+        - python  >={{ python_min }}
+    test:
+      source_files:
+        - tests
+      commands:
+        - pytest ./tests/ -v -k "not gravity" --ignore-glob=*pytorch* --ignore-glob=*CasADi* --ignore-glob=*idyntree*
+      requires:
+        - python  >={{ python_min }}
+        - pip
+        - pytest
+        - icub-models
+        - gym-ignition-models
+        - idyntree
+        - git
+        - gitpython
+        - jax
+
+  - name: {{ name|lower }}-pytorch
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python  {{ python_min }}
+      run:
+        - {{ pin_subpackage('adam-robotics', max_pin='x.x.x') }}
+        - jax2torch
+        - pytorch
+        - python  >={{ python_min }}
+    test:
+      source_files:
+        - tests
+      commands:
+        - pytest ./tests/ -v -k "not gravity" --ignore-glob=*Jax* --ignore-glob=*CasADi* --ignore-glob=*idyntree*
+      requires:
+        - python  >={{ python_min }}
```

```suggestion
        - python  {{ python_min }}
```

See https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python for more details. The idea is that you want to check in tests if the minimum supported version is actually working (so you want to pin it) while for run any version newer is ok.

---

Thanks! Addressed in https://github.com/conda-forge/adam-robotics-feedstock/pull/12/commits/59c845cd57a37d4bc6d8bfb3371bb1529c05c3a5 with rebase